### PR TITLE
[pkg/ottl] Add `Day` converter

### PR DESCRIPTION
**Pull request by sinkingpoint**
**Description:**

This adds in a `Day` converter that takes a `time.Time` and returns the int day component. Analogous to what we already have with `Hour`

**Link to tracking Issue:** #33106

**Testing:** Added two unit tests based on the existing `Hour` ones

**Documentation:** Added the new converter to the ottlfuncs README.md

### CODE REVIEW

File: pkg/ottl/ottlfuncs/README.md [hunk 1]
```diff
@@ -480,6 +481,20 @@ Examples:
 
 - `ConvertCase(metric.name, "snake")`
 
+### Day
+
+`Day(value)`
+
+The `Day` Converter returns the day component from the specified time.  The Converter [uses the `time.Day` function](https://pkg.go.dev/time#Time.Day).
+
+`value` is a `time.Time`. If `value` is another type an error is returned.
```
Comment (evan-bradley) [inline pkg/ottl/ottlfuncs/README.md:490]:
```suggestion
The `Day` Converter returns the day component from the specified time using the Go stdlib [`time.Day` function](https://pkg.go.dev/time#Time.Day).

`value` is a `time.Time`. If `value` is another type, an error is returned.
```

File: .chloggen/add-day-converter.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: pkg/ottl
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Adds converters to extract the day component of a `time.Time`
```
Comment (evan-bradley) [inline .chloggen/add-day-converter.yaml:10]:
```suggestion
note: Add the `Day` Converter to extract the int Day component from a time.Time
```

**Review by evan-bradley — APPROVED** [review]

**Comment by sinkingpoint** [comment]
Thanks for the review @evan-bradley ! I've updated it with your changes. Those build failures seem unrelated?

File: pkg/ottl/ottlfuncs/README.md [hunk 3]
```diff
@@ -480,6 +481,20 @@ Examples:
 
 - `ConvertCase(metric.name, "snake")`
 
+### Day
+
+`Day(value)`
+
+The `Day` Converter returns the day component from the specified time using the Go stdlib [time.Day` function](https://pkg.go.dev/time#Time.Day).
```
Comment (evan-bradley) [inline pkg/ottl/ottlfuncs/README.md:488]:
```suggestion
The `Day` Converter returns the day component from the specified time using the Go stdlib [`time.Day` function](https://pkg.go.dev/time#Time.Day).
```

**Comment by evan-bradley** [comment]
cc @TylerHelmuth if you want to look, this is the same as the `Year` function. If not, I'll merge this before the end of the week.

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**